04_Scheduling/03_Tasks_And_Threads.md

```diff
@@ -0,0 +1,132 @@
+# Tasks and Threads
+
+** THIS SECTION IS IN EARLY STAGES For now it will be more a set of bullet lists with things to be expaneded in the future ** 
+
+## Tasks
+
+### What are tasks
+
+In a task usually there are the following information: 
+
+* Virtual memory base
```

Comment:
I know it's early days, but I think the bare minimum for a task is just an id, list of threads and virtual memory manager. Everything else is optional, even the name.
For a thread all you really need is an id, its parent task, and the current frame. Of course it could also be useful to keep track of the stack base (so you can free it when the thread is dead).